pkg/controller/persistence/dgraph/dgraph.go

```diff
@@ -0,0 +1,100 @@
+package dgraph
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/dgraph-io/dgo"
+	"github.com/dgraph-io/dgo/protos/api"
+	"google.golang.org/grpc"
+)
+
+var Client *dgo.Dgraph
+var Connection *grpc.ClientConn
+
+func init() {
+	Open("127.0.0.1:9080")
+	err := CreateSchema()
+	if err != nil {
+		fmt.Println("Error while creating schema ", err)
+	}
+}
+
+func Open(url string) error {
+	conn, err := grpc.Dial(url, grpc.WithInsecure())
+	if err != nil {
+		return err
+	}
+
+	Connection = conn
+	dc := api.NewDgraphClient(Connection)
+	Client = dgo.NewDgraphClient(dc)
+
+	return nil
+}
+
+func Close() {
+	Connection.Close()
+}
+
+func CreateSchema() error {
+	op := &api.Operation{}
+	op.Schema = `
+		name: string @index(term) .
+		xid:  string @index(term) .
+		isService: string @index(term) .
```

Comment:
instead of two fields `isService` and `isPod` is it better to have `resourceType` which indicates type of resource?

Reply:
Refer "Nodes A Type" in the [link](https://docs.dgraph.io/howto/). Having only one variable to know the type has some serious drawbacks.